### ISSUE TITLE
Extract nginx proxy into its own package

### ISSUE THREAD

**Issue by rgbkrk** [JuliaLang/JuliaBox#193]
It would be really exciting to have the proxy work you've done here extracted into an API based proxy that we could use for JupyterHub or tmpnb, as it would be a common effort.

**Comment by ViralBShah** [comment]
+1

**Comment by tanmaykm** [comment]
Yes, that will be useful. We have been thinking about packaging the nginx proxy and JuliaBox tornado instance as docker containers, and can take up making the proxy component generic.

@rgbkrk Could you elaborate the API support you would want to have for that? Today the nginx proxy just looks at cookies to route the requests.

**Comment by rgbkrk** [comment]
The proxy we have been using, https://github.com/jupyter/configurable-http-proxy, provides a REST API to update routes on an admin port (typically `localhost:8001`).

Example:

```
POST /api/routes/[:path]
```

The overall API is [part of the README](https://github.com/jupyter/configurable-http-proxy#rest-api).

/cc @minrk
